apps/settings/tasks/ldap.py

```diff
@@ -90,9 +90,12 @@ def import_ldap_ha_user():
 
 
 def register_periodic_task(task_name, task_func, interval_key, enabled_key, crontab_key, **kwargs):
-    interval = kwargs.get(interval_key, settings.AUTH_LDAP_SYNC_INTERVAL)
-    enabled = kwargs.get(enabled_key, settings.AUTH_LDAP_SYNC_IS_PERIODIC)
-    crontab = kwargs.get(crontab_key, settings.AUTH_LDAP_SYNC_CRONTAB)
+    interval = kwargs.get(interval_key, getattr(settings, interval_key))
+    enabled = kwargs.get(enabled_key, getattr(settings, enabled_key))
+    crontab = kwargs.get(crontab_key, getattr(settings, crontab_key))
+
+    if not enabled:
+        disable_celery_periodic_task(task_name)
 
     if isinstance(interval, int):
         interval = interval * 3600
```

Comment:
Sorry, but I cannot review and analyze your code without seeing it first. Please provide me with the specific parts of the code that you would like to have reviewed.